test/attributes-option.test.js

```
@@ -172,6 +172,100 @@ describe("'attributes' option", () => {
+    ).toMatchSnapshot('result');
+    expect(getWarnings(stats)).toMatchSnapshot('warnings');
+    expect(getErrors(stats)).toMatchSnapshot('errors');
+  });
+
+  it('should handle all src attributes in all HTML tags when tag is empty', async () => {
+    const compiler = getCompiler('simple.js', {
+      attributes: {
+        list: [
+          {
+            tag: '',
+            attribute: 'src',
+            type: 'src',
+          },
+        ],
+      },
+    });
+    const stats = await compile(compiler);
+
+    expect(getModuleSource('./simple.html', stats)).toMatchSnapshot('module');
+    expect(
+      execute(readAsset('main.bundle.js', compiler, stats))
+    ).toMatchSnapshot('result');
+    expect(getWarnings(stats)).toMatchSnapshot('warnings');
+    expect(getErrors(stats)).toMatchSnapshot('errors');
+  });
+
+  it('should handle all src attributes in all HTML tags except img (testing filter option) tag is undefined', async () => {
+    const compiler = getCompiler('simple.js', {
+      attributes: {
+        list: [
+          {
+            attribute: 'src',
+            type: 'src',
+            // eslint-disable-next-line no-unused-vars
+            filter: (tag, attribute, attributes) => {
+              return tag.toLowerCase() !== 'img';
+            },
+          },
+        ],
+      },
+    });
+    const stats = await compile(compiler);
+
+    expect(getModuleSource('./simple.html', stats)).toMatchSnapshot('module');
+    expect(
+      execute(readAsset('main.bundle.js', compiler, stats))
+    ).toMatchSnapshot('result');
+    expect(getWarnings(stats)).toMatchSnapshot('warnings');
+    expect(getErrors(stats)).toMatchSnapshot('errors');
+  });
+
+  it('should handle all src attributes in all HTML tags except img (testing filter option) tag is an empty string', async () => {
+    const compiler = getCompiler('simple.js', {
+      attributes: {
+        list: [
+          {
+            tag: '',
```

Should throw an error